### PR TITLE
Backport #5850

### PR DESCRIPTION
**Pull request by no-reply**
limited/guarded backport of query portion; following this up with deprecations, test updates, etc...

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by cjcolvar** [comment]
This backport PR was created before the upstream PR got merged and I think it is missing some of the work that was incorporated later.  Specifically this commit: https://github.com/samvera/hyrax/pull/5850/commits/78a6098d8ad9921415d5951d0d32600a018723d6

**Comment by orangewolf** [comment]
@no-reply @cjcolvar this has everything but the max depth in it. Is that intentional? I'd love to get this out of draft and merged in to a 3.x release. If it needs the max depth I can do that. If it doesn't then can we undraft and merge?

**Comment by ShanaLMoore** [comment]
Tamsin confirmed that max depth was not left out on purpose. Rob did a line by line comparison. He will follow up by adding it.

**Comment by orangewolf** [comment]
@cjcolvar I went back and compared the final merged version w/ this code. the maxdepth was missing, but I don't see any other missing changes. Does it look like it has parity to you now?

**Review by cjcolvar — COMMENTED** [review]
This looks good, but it would probably be good to include some tests to ensure the opt-in solr graph path works as expected.

File: app/search_builders/hyrax/dashboard/nested_collections_search_builder.rb [hunk 1]
```diff
@@ -26,11 +26,18 @@ def with_pagination(solr_parameters)
...
-        solr_parameters[:fq] += [
-          "-" + Hyrax::SolrQueryBuilderService.construct_query_for_ids([limit_ids]),
-          Hyrax::SolrQueryBuilderService.construct_query(Hyrax.config.collection_type_index_field => @collection.collection_type_gid)
-        ]
-        solr_parameters[:fq] += limit_clause if limit_clause # add limits to prevent illegal nesting arrangements
+        if Hyrax.config.use_solr_graph_for_collection_nesting
+          solr_parameters[:fq] += [
+            "-{!graph from=id to=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_parent}}id:#{@collection.id}",
+            "-{!graph to=id from=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_child}}id:#{@collection.id}"
+          ]
```
Comment (cjcolvar) [inline app/search_builders/hyrax/dashboard/nested_collections_search_builder.rb:33]: Looks like the first clause got missed.
```suggestion
          solr_parameters[:fq] += [
            Hyrax::SolrQueryBuilderService.construct_query(Hyrax.config.collection_type_index_field => @collection.collection_type_gid),
            "-{!graph from=id to=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_parent}}id:#{@collection.id}",
            "-{!graph to=id from=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_child}}id:#{@collection.id}"
          ]
```

**Review by cjcolvar — COMMENTED** [review]
What about adding a loop over all of the form tests (`spec/forms/hyrax/forms/dashboard/nest_collection_form_spec.rb`) to check they work with nested indexing enabled and disabled?
I think that along with the tests already added should cover testing the exact query and the user-facing functionality.

**Comment by orangewolf** [comment]
> What about adding a loop over all of the form tests (`spec/forms/hyrax/forms/dashboard/nest_collection_form_spec.rb`) to check they work with nested indexing enabled and disabled?
> I think that along with the tests already added should cover testing the exact query and the user-facing functionality.

that makes sense. I likely cant do it today, but I'll see if can get to it tomorrow or more likely Friday.

**Review by cjcolvar — APPROVED** [review]
Looks great!  Thanks for backporting this!

**Pull request merged**